### PR TITLE
[SNOW-675611] Get ErrorCode enum from SFException

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Get the ErrorCode for Exception handling in Kafka Connector

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/utils/SFException.java [hunk 1]
```diff
@@ -58,4 +60,8 @@ public Object[] getParams() {
   public Throwable getCause() {
     return cause;
   }
+
+  public ErrorCode getErrorCode() {
```
Comment (sfc-gh-japatel) [inline src/main/java/net/snowflake/ingest/utils/SFException.java:64]: can you add a comment here saying this is error code returned from snowflake and nothing we created here?

**Review by sfc-gh-japatel — APPROVED** [review]
lgtm! Thanks

File: src/main/java/net/snowflake/ingest/utils/SFException.java [hunk 2]
```diff
@@ -18,6 +18,7 @@ public class SFException extends RuntimeException {
   private Throwable cause;
   private String vendorCode;
   private Object[] params;
+  private ErrorCode errorCode;
```
Comment (sfc-gh-tzhang) [inline src/main/java/net/snowflake/ingest/utils/SFException.java:21]: Then we should be able to remove vendorCode?

**Comment by sfc-gh-rcheng** [comment]
further investigation showed that this change is unnecessary

**Pull request closed without merging**